packages/core/src/utils/guards.ts

```diff
@@ -131,3 +132,23 @@ export function isCallable(
 
   return future.type === "contract" || future.type === "library";
 }
+
+export function isContract(
+  future: DeploymentGraphFuture
+): future is ContractFuture {
+  if (isProxy(future)) {
```

Comment:
Does this need to be recursive? Just saying it makes me wince.

Reply:
Honestly, no idea, I just copied it from the `isCallable` function above it

Reply:
I will add a linear ticket with a potential test (passing params through nested modules so param wrapped in param).